repo/manifest/committed_manifest_manager.go

```diff
@@ -30,17 +33,23 @@ type committedManifestManager struct {
 	// +checklocks:cmmu
 	locked bool
 	// +checklocks:cmmu
-	committedEntries map[ID]*manifestEntry
+	committedEntries map[ID]*inMemManifestEntry
 	// +checklocks:cmmu
 	committedContentIDs map[content.ID]bool
 
 	// autoCompactionThreshold controls the threshold after which the manager auto-compacts
 	// manifest contents
 	// +checklocks:cmmu
 	autoCompactionThreshold int
+
+	// formatVersion is the serialization version that these manifests have.
+	// Version 0 stored all manifest content inline with metadata. Version 1 adds
+	// a level of indirection to store manifest content as separate content blobs
+	// and has the metadata point to the content blob.
+	formatVersion int
```

Comment:
I presume not all repositories want this behavior, as it will create a lot more contents, which has its downsides. we would need to have a way to enabling this behavior per repository.

Reply:
> allows users to set a flag and only upgrade to this way of storing manifests if requested

What is the upgrade like? For an existing repo, once the flag is set, will we read all the manifests, reorganize them and write them back? Will this be done through a dedicate CLI?

Reply:
Is it possible we allow the both formats coexist in one repo? E.g., the manifests are stored inline by default, and only if there are too much of them, we compact them into a content and store them indirectly through the repo maintenance?

This would avoid the repo upgrade, or any back compatibility issues, or the downsides involved by the increase of contents as mentioned above.

Reply:
> What is the upgrade like? For an existing repo, once the flag is set, will we read all the manifests, reorganize them and write them back? Will this be done through a dedicate CLI?

This hasn't been hashed out completely yet, but yes, likely read in all manifests (already done anyway when populating the manifest manager) and write them in the new format. This could technically either be a dedicated operation or happen during manifest compaction if the user has requested an upgrade. As the new format isn't completely compatible with the older one, doing it as a separate upgrade operation would be safer

> Is it possible we allow the both formats coexist in one repo? E.g., the manifests are stored inline by default, and only if there are too much of them, we compact them into a content and store them indirectly through the repo maintenance?

yes, this would be possible. One of the things jkowalski mentioned out of band is to continue to write all manifests in the old format and compact them into the newer format and/or only use the new format if we had more than a certain number of manifests total